modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTCollectionHistoryProviderRegistryImpl.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.change.tracking.service.impl;
+
+import com.liferay.change.tracking.spi.history.CTCollectionHistoryProvider;
+import com.liferay.change.tracking.spi.history.CTCollectionHistoryProviderRegistry;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Brooke Dalton
+ */
+@Component(service = CTCollectionHistoryProviderRegistry.class)
+public class CTCollectionHistoryProviderRegistryImpl
+	implements CTCollectionHistoryProviderRegistry {
+
+	@Override
+	public CTCollectionHistoryProvider<?> getCTCollectionHistoryProvider(
+		String classNameId) {
+
+		return _serviceTrackerMap.getService(classNameId);
```

Comment:
we need to get the className from the given classNameId. Use `classNameLocalService.fetchByClassNameId(classNameId)` to get the className

Reply:
Do I need to check for a null className?

Reply:
Not necessary. If you pass in a null key, it will return null.

Reply:
Actually, let's do a null check since it returns a ClassName object, and we need to call getValue on the object.